slackbridge/bots.py

```diff
@@ -67,39 +130,30 @@ def post_to_slack(self, user, channel, message):
 
     def check_slack_rtm(self):
         try:
-            message = self.sc.rtm_read()
+            message_list = self.sc.rtm_read()
         except TimeoutError:
             log.err('Retrieving message from Slack RTM timed out')
-            message = None
-
-        if not message:
-            return
-
-        message = message[0]
-        log.msg(message)
-
-        if 'type' not in message:
-            return
-
-        if (message['type'] == 'presence_change' and
-                message['user'] in self.users):
-            user_bot = self.users[message['user']]
-            if message['presence'] == 'away':
-                user_bot.away('Slack user inactive.')
-            elif message['presence'] == 'active':
-                user_bot.back()
             return
 
-        if (message['type'] != 'message' or
-                'user' not in message or
-                'bot_id' in message):
+        if not message_list:
             return
 
-        if (message['user'] in self.users and
-                message['channel'] in self.channels):
-            user_bot = self.users[message['user']]
-            channel = self.channels[message['channel']]
-            return user_bot.post_to_irc('#' + channel['name'], message['text'])
+        for message in message_list:
+            log.msg(message)
+
+            if 'type' in message:
+                message_obj = self.SlackMessage(
+                    message, self.user_bots, self.channels)
+                self.message_queue.put(message_obj)
+
+    def empty_queue(self):
+        while not self.message_queue.empty():
+            try:
+                message = self.message_queue.get_nowait()
+            except queue.Empty:
+                return deferLater(Clock(), 0, self.message_queue.join)
+            message.resolve()
+            self.message_queue.task_done()
```

Comment:
I also think a comment might be useful here, since normally I wouldn't expect a queue to need to say when it is done. Maybe some details from [the docs](https://docs.python.org/3/library/queue.html#queue.Queue.task_done)?